mule-extensions-api-xml-dsl/src/test/java/org/mule/runtime/extension/xml/dsl/test/BaseXmlDeclarationTestCase.java

```diff
@@ -101,16 +105,32 @@ public void before() {
 
     when(source.getName()).thenReturn(SOURCE_NAME);
     when(operation.getName()).thenReturn(OPERATION_NAME);
+    when(operation.getModelProperty(ConfigTypeModelProperty.class)).thenReturn(empty());
     when(configuration.getName()).thenReturn(CONFIGURATION_NAME);
     when(connectionProvider.getName()).thenReturn(CONNECTION_PROVIDER_NAME);
 
+    noConfigOn(connectionProvider, operation, source, configuration);
+    noConnectionOn(connectionProvider, operation, source, configuration);
     when(dslContext.getExtension(any())).thenReturn(empty());
 
     Stream.of(configuration, operation, connectionProvider, source).forEach(
                                                                             model -> when(model.getParameterModels())
                                                                                 .thenReturn(asList(parameterModel)));
   }
 
+  private void noConfigOn(EnrichableModel... models) {
+    for (EnrichableModel model : models) {
+      when(model.getModelProperty(ConfigTypeModelProperty.class)).thenReturn(empty());
+    }
+  }
+
+  private void noConnectionOn(EnrichableModel... models) {
+    for (EnrichableModel model : models) {
+      when(model.getModelProperty(ConnectivityModelProperty.class)).thenReturn(empty());
+      when(model.getModelProperty(PagedOperationModelProperty.class)).thenReturn(empty());
```

Comment:
the paging delegate is fed with a connection each time it fetches a page, and it does not require the operation to receive a `@Connection` parameter. Therefore, it's connected